crates/rolldown/src/utils/chunk/finalize_chunks.rs

```diff
@@ -121,9 +121,11 @@ pub fn finalize_assets(
       // TODO: PERF: should check if this asset has dependencies/placeholders to be replaced
       match &mut asset.content {
         StrOrBytes::Str(content) => {
-          *content =
-            replace_placeholder_with_hash(mem::take(content), &final_hashes_by_placeholder)
-              .into_owned();
+          if let std::borrow::Cow::Owned(str) =
```

Comment:
Maybe it's better to hide this logic inside the `replace_placeholder_with_hash` function. It would be confusing for people who doesn't understand how `regex` works.

I think the original `replace_placeholder_with_hash` is good enough for this use case. Though it still create a dummy `String::default`.

Reply:
This is relatively subjective to say which one is better, previous impl still does not hide full implementation inside the function, you need to take `content` and then convert it from `Cow` to `String` after processed.

Reply:
Yeah. Thanks for the explanation. It's indeed quite hard to dealing with due to `&T`, `T` and the ownership of rust.